packages/components/src/components/ViewYAML/ViewYAML.js

```diff
@@ -15,29 +15,54 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import jsYaml from 'js-yaml';
 import classNames from 'classnames';
+import { Prism as SyntaxHighlight } from 'react-syntax-highlighter';
 
 const ViewYAML = props => {
-  const { className, dark, resource } = props;
-
-  return (
-    <div
-      className={classNames('bx--snippet--multi', className, {
-        'tkn--view-yaml--dark': dark
-      })}
-    >
-      <code>
-        <pre>{jsYaml.dump(resource)}</pre>
-      </code>
-    </div>
-  );
+  const { className, dark, resource, enableSyntaxHighlighting = false } = props;
+
+  const renderRaw = (clz, yaml) => {
+    return (
+      <div>
+        <code className={clz}>
+          <pre>{yaml}</pre>
+        </code>
+      </div>
+    );
+  };
+
+  const renderSyntaxHighlight = (clz, yaml) => {
+    return (
+      <SyntaxHighlight
+        className={clz}
+        language="yaml"
+        useInlineStyles={false}
+        codeTagProps={{}}
+      >
+        {yaml}
+      </SyntaxHighlight>
+    );
+  };
```

Comment:
Since these functions receive all of their required inputs as params anyway, they can be defined outside the component to avoid unnecessarily redefining them on every render

They could also be modified slightly to be their own React components:

```
function YAMLHighlighter({ children, className }) {
  return (
    <SyntaxHighlight
      className={className}
      language="yaml"
      useInlineStyles={false}
      codeTagProps={{}}
    >
      {children}
    </SyntaxHighlight>
  );
}

function ViewYAML(props) {
…
  if (enableSyntaxHighlighting) {
    return <YAMLHighlighter className={className}>{yaml}</YAMLHighlighter>;
  }
…
};
```

Reply:
@AlanGreene The render functions already extracted and I change `ViewYAML` component to be function instead of const variable. Please kindly review again.